persistence/jdbc/src/test/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStoreTest.java

```diff
@@ -50,61 +37,24 @@ protected String parameters() {
-      storeBuilder.segmented(segmented);
-      UnitTestDatabaseManager.configureUniqueConnectionFactory(storeBuilder);
-      UnitTestDatabaseManager.buildTableManipulation(storeBuilder.table());
-      JdbcStringBasedStore stringBasedCacheStore = new JdbcStringBasedStore();
-      stringBasedCacheStore.init(createContext(builder.build()));
-      return stringBasedCacheStore;
+   protected JdbcStringBasedStore createStore() throws Exception {
+      return new JdbcStringBasedStore();
    }
 
-   public void testNotCreateConnectionFactory() throws Exception {
-      ConfigurationBuilder builder = TestCacheManagerFactory
-            .getDefaultCacheConfiguration(false);
-      JdbcStringBasedStoreConfigurationBuilder storeBuilder = builder
+   @Override
+   protected Configuration buildConfig(ConfigurationBuilder configurationBuilder) {
+      JdbcStringBasedStoreConfigurationBuilder storeBuilder = configurationBuilder
             .persistence()
-            .addStore(JdbcStringBasedStoreConfigurationBuilder.class)
-            .manageConnectionFactory(false)
-            .dialect(DatabaseType.H2)
-            .dbMajorVersion(1)
-            .dbMinorVersion(4);
-
+            .addStore(JdbcStringBasedStoreConfigurationBuilder.class);
+      storeBuilder.segmented(segmented);
+      UnitTestDatabaseManager.configureUniqueConnectionFactory(storeBuilder);
       UnitTestDatabaseManager.buildTableManipulation(storeBuilder.table());
-      storeBuilder.table()
-            .createOnStart(false);
-
-      JdbcStringBasedStore stringBasedCacheStore = new JdbcStringBasedStore();
-      stringBasedCacheStore.init(createContext(builder.build()));
-
-      // this will make sure that if a method like stop is called on the connection then it will barf an exception
-      ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
-      TableManager tableManager = mock(TableManager.class);
-      TableManager.Metadata meta = new AbstractTableManager.MetadataImpl(Version.getVersionShort(), HashConfiguration.NUM_SEGMENTS.getDefaultValue());
-      when(tableManager.metaTableExists(null)).thenReturn(true);
-      when(tableManager.getMetadata(null)).thenReturn(meta);
-
-      tableManager.start();
-      assertNull(stringBasedCacheStore.getConnectionFactory());
-      stringBasedCacheStore.setConnectionFactory(connectionFactory);
-      stringBasedCacheStore.setTableManager(tableManager);
-      stringBasedCacheStore.start();
-      assertSame(stringBasedCacheStore.getConnectionFactory(), connectionFactory);
-
-      //stop should be called even if this is an external
-      reset(tableManager, connectionFactory);
-      tableManager.stop();
-
-      stringBasedCacheStore.stop();
+      return configurationBuilder.build();
    }
 
    @Override
    protected boolean storePurgesAllExpired() {
-      // expiration listener is not called for the entries
+      // TODO: we need to add back in expiration support
```

Comment:
Don't forget :slightly_smiling_face:

Reply:
I actually implemented it! I however did not change this test as the new expiration code works properly to expose the entry unlike the old expiration code. Let me see if just removing this method works, heh.